web/src/stream_popover.ts

```diff
@@ -440,18 +461,10 @@ export async function build_move_topic_to_stream_popover(
             select_stream_id = stream_widget_value;
         }
 
-        let {
-            current_stream_id,
-            new_topic_name,
-            send_notification_to_new_thread,
-            send_notification_to_old_thread,
-        } = params;
-        send_notification_to_new_thread = send_notification_to_new_thread === "on";
-        send_notification_to_old_thread = send_notification_to_old_thread === "on";
-        current_stream_id = Number.parseInt(current_stream_id, 10);
-        if (select_stream_id !== undefined) {
-            select_stream_id = Number.parseInt(select_stream_id, 10);
-        }
+        let new_topic_name = params.new_topic_name;
+        const send_notification_to_new_thread = params.send_notification_to_new_thread === "on";
+        const send_notification_to_old_thread = params.send_notification_to_old_thread === "on";
+        const current_stream_id = Number.parseInt(params.current_stream_id, 10);
```

Comment:
I think this lost a `parseInt`  on `stream_widget_value`? The weirdness seems to be that the `select_stream_id` parsing logic should have been in the `if/else` above where it is set.

Reply:
It was exchanged for an added `parseInt` in `move_topic_on_update`, making `stream_widget_value` always a number (to match `stream_widget_value = current_stream_id`).